code/game/objects/items/explosives/mine.dm

```diff
@@ -63,7 +99,9 @@
 /obj/item/explosive/mine/attack_self(mob/living/user)
 	if(!..())
 		return
-
+	if(needs_digging && user.loc && (user.loc.density)) //  || is_mainship_level(user.z)
```

Comment:
- comment code bad
- why is there a density check
- it shouldn't be wrapped in ()